src/cargo/util/context/mod.rs

```diff
@@ -681,7 +681,17 @@ impl GlobalContext {
                 }),
             ];
 
-            let path = val.resolve_templated_path(self, replacements);
+            let path = val
+                .resolve_templated_path(self, replacements)
+                .map_err(|e| match e {
+                    path::ResolveTemplateError::UnexpectedVariable {
+                        variable,
+                        raw_template,
+                    } => anyhow!(
+                        "Unexpected variable `{variable}` in build.build-dir path `{raw_template}`"
```

Comment:
We try to follow the [rustc style guide](https://rustc-dev-guide.rust-lang.org/diagnostics.html#diagnostic-output-style-guide) which says this error should be lower case